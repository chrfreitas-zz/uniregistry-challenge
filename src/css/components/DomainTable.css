
.is-editing .domain-table {
    opacity: 0;
    top: -50%;
    transform: translateY(50%);
}

.domain-table {
    transition: opacity .5s, transform .5s;
    background-color: var(--blue);
    padding: 35px 15px;
    border-radius: 5px;
    color: var(--white);
    opacity: 1;
    transform: translateY(0px);
}
