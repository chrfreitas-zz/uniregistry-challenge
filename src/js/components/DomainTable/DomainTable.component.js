import React, { Component } from 'react';
import PropTypes from 'prop-types';

class DomainTable extends Component {
    static propTypes = {
        domains: PropTypes.array,
        goEdit: PropTypes.func
    }

    handleClickRow = (domain) => {
        this.props.goEdit(domain);
    }

    renderHeader() {
        return (
            <tr>
                <th>Domains name</th>
                <th>Uniregistry</th>
                <th>Price</th>
            </tr>
        )
    }

    renderBody(domains = []) {
        return domains.map((domain) => {
            return (
                <tr key={domain.id} onClick={ this.handleClickRow.bind(this, domain) }>
                    <td>{ domain.description }</td>
                    <td>{ domain.uniregistry }</td>
                    <td>{ domain.price }</td>
                </tr>
            );
        });
    }

    render() {
        return (
            <table className="table table-hover">
                <thead>
                    { this.renderHeader() }
                </thead>
                <tbody>
                    { this.renderBody(this.props.domains) }
                </tbody>
            </table>
        )
    }
}

export default DomainTable;