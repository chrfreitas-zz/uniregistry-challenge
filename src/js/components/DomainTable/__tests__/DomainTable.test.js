import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import DomainTable from '../DomainTable.component';

beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
});

describe('DomainTable component', () => {
    it('It should match with the snapshot', () => {
        const wrapper = renderer.create(<DomainTable />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('It should render a html list', () => {
        const domains = [
            {
                id: 1,
                name: 'foofighter.lol',
                uniregistry: 'Yes',
                price: '$12.00'
            }
        ];

        const wrapper = shallow(<DomainTable domains={domains} />);

        const amountDomains = domains.length;
        const amountDomainsRendered = wrapper.find('tbody tr').length;

        expect(amountDomains).toEqual(amountDomainsRendered);
    });
});
