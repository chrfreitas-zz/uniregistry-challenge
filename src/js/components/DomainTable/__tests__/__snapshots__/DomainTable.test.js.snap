// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainTable component It should match with the snapshot 1`] = `
<table
  className="table table-hover"
>
  <thead>
    <tr>
      <th>
        Domains name
      </th>
      <th>
        Uniregistry
      </th>
      <th>
        Price
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
